avm/res/kusto/cluster/database/principal-assignment/main.bicep

```
@@ -0,0 +1,61 @@
+metadata name = 'Kusto Cluster Database Principal Assignments'
+metadata description = 'This module deploys a Kusto Cluster Database Principal Assignment.'
+
+@minLength(4)
+@maxLength(22)
+@description('Conditional. The name of the parent Kusto Cluster Database. Required if the template is used in a standalone deployment.')
+param kustoClusterDatabaseName string
+
+@description('Required. The principal assignement for the Kusto Cluster Database.')
+param databasePrincipalAssignment databasePrincipalAssignmentType
+
+// ============== //
+// Resources      //
+// ============== //
+
+resource kustoClusterDatabase 'Microsoft.Kusto/clusters/databases@2024-04-13' existing = {
+  name: kustoClusterDatabaseName
+}
+
+resource kustoDatabasePrincipalAssignment 'Microsoft.Kusto/clusters/databases/principalAssignments@2024-04-13' = {
+  name: databasePrincipalAssignment.principalId
+  parent: kustoClusterDatabase
+  properties: {
+    principalId: databasePrincipalAssignment.principalId
+    principalType: databasePrincipalAssignment.principalType
+    role: databasePrincipalAssignment.role
+    tenantId: databasePrincipalAssignment.tenantId
+  }
+}
+
+// =============== //
+//     Outputs     //
+// =============== //
+
+@description('The name of the deployed Kusto Cluster Database Principal Assignment.')
+output name string = kustoDatabasePrincipalAssignment.name
+
+@description('The resource id of the deployed Kusto Cluster Database Principal Assignment.')
+output resourceId string = kustoClusterDatabase.id
+
+@description('The resource group name of the deployed Kusto Cluster Database Principal Assignment.')
+output resourceGroupName string = resourceGroup().name
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+@export()
+type databasePrincipalAssignmentType = {
```

Couldn't help it, sorry.
```suggestion
@description('The type of a database principal assignment.')
type databasePrincipalAssignmentType = {
```